.github/workflows/scripts/download-nvd.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+
+# This script fetches NVD CVE data for a given year, verifies its integrity, and uploads it to a GCS bucket.
+
+# ------------------------ Instructions ------------------------
+# 1. Ensure you have `curl`, `gunzip`, and `gsutil` installed.
+# 2. Make this script executable: `chmod +x <script_name>.sh`
+# 3. Run this script with a year as an argument: `./<script_name>.sh 2023`
+# 4. Check the script logs for any errors. The script will stop execution if there's an error.
+
+# Function to download a file with curl and handle errors
+download_file() {
+    local url=$1
+    local output_path=$2
+
+    curl --fail --silent --show-error --max-time 60 --retry 3 -o "$output_path" "$url" || (echo "Error fetching file from $url" && exit 1)
```

Comment:
I don't think this `exit 1` will work as expected, because you essentially created a sub-shell using `(` and `)`.